lightning/src/ln/onion_message.rs

```diff
@@ -9,6 +9,25 @@
 
 //! Onion Messages: sending, receiving, forwarding, and ancillary utilities live here
 
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) struct Packet {
+	pub(crate) version: u8,
+	/// We don't want to disconnect a peer just because they provide a bogus public key, so we hold a
+	/// Result instead of a PublicKey as we'd like.
+	pub(crate) public_key: Result<PublicKey, secp256k1::Error>,
```

Comment:
Haven't looked at the full context of where this is constructed / used, but could that return a `Err` if the public key is bad?

Reply:
Here's the context: https://github.com/lightningdevkit/rust-lightning/pull/1432#discussion_r854129322 lmk if more clarification is desired